### PR TITLE
refactor: remove unused partial init from cli deploy

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
After https://github.com/anza-xyz/agave/pull/877 buffers are always validated before deploys so there's no need to have extra logic for partial initialization of buffers anymore

#### Summary of Changes
- Remove `complete_partial_program_init` function
- Remove `allow_excessive_balance` usage since it was only used in the above function
- Document `--allow-excessive-balance` as obsolete

Fixes #

### CODE REVIEW

File: cli/src/program.rs [hunk 1]
```diff
@@ -248,7 +245,7 @@ impl ProgramSubCommands for App<'_, '_> {
                                 .takes_value(false)
                                 .help(
                                     "Use the designated program id even if the account already \
-                                     holds a large balance of SOL",
+                                     holds a large balance of SOL (Obsolete)",
```
Comment (joncinque) [inline cli/src/program.rs:249]: Can you add the `hidden()` flag on this?

**Review by joncinque — COMMENTED** [review]
Just the one comment, looks great otherwise!

**Review by joncinque — DISMISSED** [review]

**Comment by jstarry** [comment]
rebased due to upstream conflict

**Review by joncinque — DISMISSED** [review]

**Comment by jstarry** [comment]
rebased again to fix another conflict

**Review by joncinque — APPROVED** [review]

**Pull request merged**